haystack/pipelines/base.py

```diff
@@ -690,6 +692,80 @@ def run(  # type: ignore
                 i += 1  # attempt executing next node in the queue as current `node_id` has unprocessed predecessors
         return node_output
 
+    @classmethod
+    def eval_beir(
+        cls,
+        index_pipeline: Pipeline,
+        query_pipeline: Pipeline,
+        index_params: dict = {},
+        query_params: dict = {},
+        dataset: str = "scifact",
+        dataset_dir: Path = Path("."),
+        top_k_values: List[int] = [1, 3, 5, 10, 100, 1000],
+    ) -> Tuple[Dict[str, float], Dict[str, float], Dict[str, float], Dict[str, float]]:
+        """
+        Runs information retrieval evaluation of a pipeline using BEIR on a specified BEIR dataset.
+        See https://github.com/beir-cellar/beir for more information.
```

Comment:
Maybe it's worth mentioning that an index beir_{dataset} will be created/deleted. At first, I was afraid that users will delete their already indexed data easily when trying this new feature out. Later on I learned that there is no such risk.

Reply:
I had to change this again since index creation and thus ensuring appropriate field mappings is only done at document store init time. So I got rid of the dedicated index and leave it up to the user. If the user provides a non-empty index a HaystackError will be thrown.